application-forum-ui/src/main/resources/ForumCode/Actions.xml

```diff
@@ -65,26 +67,34 @@
         $response.setStatus(403)
       #end
     #end
-  #elseif("$!request.action" == "addtopic")
+  #elseif("$!request.action" == 'addtopic')
     #if("$!request.title.trim()" != '')
       #if("$!request.space" != '')
-        #set($newTopicName = $request.page)
         #set($topicTitle = "$!request.title.trim()")
-        #set($topicDoc = $xwiki.getDocument($newTopicName))
+        #set($topicSpace = "$!request.space")
+        #if($supportsNS)
+          #set($topicSpaceReference = $services.model.resolveSpace($topicSpace))
```

Comment:
Careful here. If `$topicSpace` is user-entered content and the user enters `My topic with a . in the name` resolveSpace will see 2 spaces there (`My topic with a ` and the subspace ` in the name`). What you were probably looking for was `$services.model.createSpaceReference($topicSpace, $doc.documentReference.wikiReference)`. This way `$topicSpace` is treated as a space name and not as a space reference.